CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/state/DataAccessPreferenceKeysNGTest.java

```
@@ -0,0 +1,676 @@
+     */
+    @Test
+    public void testIsSaveResultsEnabled() {
+        System.out.println("isSaveResultsEnabled");
+
+        // mocking code from another test method
+        final String key = SAVE_DATA_DIR_PREF;
+        File tempFile = null;
+        try {
+            tempFile = File.createTempFile("testfile", ".txt");
+
+            final String path = tempFile == null ? "" : tempFile.getAbsolutePath();
+
+            prefs = mock(Preferences.class);
+            when(prefs.get(Mockito.eq(key), Mockito.anyString())).thenReturn(path);
+
+            // Create static mock of NbPreferences to return the preferences mock
+            try (MockedStatic<NbPreferences> mockedStatic = Mockito.mockStatic(NbPreferences.class)) {
+                mockedStatic.when(() -> NbPreferences.forModule(Mockito.eq(DataAccessPreferenceKeys.class))).thenReturn(prefs);
+                mockedStatic.verify(() -> NbPreferences.forModule(Mockito.eq(DataAccessPreferenceKeys.class)), Mockito.never());
+
+                final File result = DataAccessPreferenceKeys.getDir(key);
+                mockedStatic.verify(() -> NbPreferences.forModule(Mockito.eq(DataAccessPreferenceKeys.class)), times(1));
+
+                // Verify mock is working correctly
+                assertEquals(result, tempFile);
+
+                // Verify no file is returned because it needs to be a directory
+                final File expResult = null;
+                assertEquals(DataAccessPreferenceKeys.getDataAccessResultsDir(), expResult);
+
+                // Test saveResultsEnabled with invalid DataAccessResultsDirectory
+                boolean expResult2 = false;
+                boolean result2 = DataAccessPreferenceKeys.isSaveResultsEnabled();
+                assertEquals(result2, expResult2);
+            }
+        } catch (final IOException ex) {
+            // Fail when test cannot create a file
+            Exceptions.printStackTrace(ex);
+            fail();
```

I would just throw the exception